src/__tests__/ParseUser-test.js

```diff
@@ -360,4 +361,49 @@ describe('ParseUser', () => {
       done();
     });
   }));
+
+  it('can get error when recursive _linkWith call fails', asyncHelper((done) => {
+    CoreManager.setRESTController({
+      request(method, path, body, options) {
+        expect(method).toBe('POST');
+        expect(path).toBe('users');
+        expect(body.authData.test).toEqual({
+          id : 'id',
+          access_token : 'access_token'
+        });
+        var error = new ParseError(
+          ParseError.ACCOUNT_ALREADY_LINKED,
+          'Another user is already linked to this facebook id.'
+        );
+        return ParsePromise.error(error);
+      },
+      ajax() {}
+    });
+    var provider = {
+      authenticate(options) {
+        if (options.success) {
+          options.success(this, {
+            id: 'id',
+            access_token: 'access_token'
+          });
+        }
+      },
+
+      restoreAuthentication(authData) {},
+
+      getAuthType() {
+        return 'test';
+      },
+
+      deauthenticate() {}
+    };
+    
+    ParseUser.logInWith(provider, {}).then(() => {
```

Comment:
you can write `.then(null, (error) => {})` instead if you don't intend to do anything with the first argument